File: DSA/LeetCode/Math/1963. Minimum Number of Swaps to Make the String Balanced.js
/**
 * @param {string} s
 * @return {number}
 */
var minSwaps = function (s) {
    let o = 0
    let c = 0
    for (let i = 0; i < s.length; i++) {

        if (s[i] == "]") {
            if (o > 0)
                o--
            else
                c++
        } else if (s[i] == "[") {
            o++
        }
    }
    return Math.ceil(c / 2)
};

File: DSA/LeetCode/Math/1963. Minimum Number of Swaps to Make the String Balanced.py
class Solution:
    def minSwaps(self, s: str) -> int:
        o = 0
        c = 0
    
        for char in s:

            if char == "]":
                if o > 0:
                    o = o-1
                else:
                    c += 1
            elif char == "[":
                o +=1    
    
        return math.ceil(c / 2)